### PR TITLE
Fixed Issue #65: Markers attach to an XMachine or an XContext 

### PR DESCRIPTION
**Pull request by tshoang**
The markers are now attached to the name of the XMachine or XContext if there are no element/feature found corresponding to the original issue

### CODE REVIEW

File: ac.soton.xeventb.xmachine/xtend-gen/ac/soton/xeventb/xmachine/validation/XMachineValidator.java [hunk 1]
```diff
@@ -309,36 +310,87 @@ private EStructuralFeature getFeature(final EObject obj, final IAttributeType at
...
-        String _plus = (message + " (from ");
-        String _plus_1 = (_plus + Integer.valueOf(start));
-        String _plus_2 = (_plus_1 + " to ");
-        String _plus_3 = (_plus_2 + Integer.valueOf(end));
-        String _plus_4 = (_plus_3 + ")");
-        message = _plus_4;
+      EStructuralFeature feature = this.getFeature(obj, attributeType);
+      EObject elem = obj;
+      if ((feature == null)) {
+        elem = this.getMachine(obj);
```
Comment (cfsnook) [inline ac.soton.xeventb.xmachine/xtend-gen/ac/soton/xeventb/xmachine/validation/XMachineValidator.java:319]: could you use 
((EventBObject)obj).getContaining(MachinePackage.Literals.MACHINE);

File: ac.soton.xeventb.xcontext/xtend-gen/ac/soton/xeventb/xcontext/validation/XContextValidator.java [hunk 2]
```diff
@@ -269,29 +271,39 @@ private Axiom findAxiom(final Context ctx, final String label) {
...
-        String _plus = (message + " (from ");
-        String _plus_1 = (_plus + Integer.valueOf(start));
-        String _plus_2 = (_plus_1 + " to ");
-        String _plus_3 = (_plus_2 + Integer.valueOf(end));
-        String _plus_4 = (_plus_3 + ")");
-        message = _plus_4;
+      EStructuralFeature feature = this.getFeature(obj, attributeType);
+      EObject elem = obj;
+      if ((feature == null)) {
+        elem = this.getContext(obj);
```
Comment (cfsnook) [inline ac.soton.xeventb.xcontext/xtend-gen/ac/soton/xeventb/xcontext/validation/XContextValidator.java:280]: could you use
((EventBObject)obj).getContaining(ContextPackage.Literals.CONTEXT);

File: ac.soton.xeventb.xcontext/src/ac/soton/xeventb/xcontext/validation/XContextValidator.xtend [hunk 3]
```diff
@@ -232,24 +234,34 @@ class XContextValidator extends AbstractXContextValidator {
...
-		val end = RodinMarkerUtil.getCharEnd(rodinMarker)
-		var message = attributes.get(IMarker.MESSAGE)
-		if (start != -1) {
-			message = message + " (from " + start + " to " + end + ")"
+        var message = attributes.get(IMarker.MESSAGE)
+		
+        var feature = getFeature(obj, attributeType)
+        var elem = obj
+        if (feature === null) {
+            elem = getContext(obj)
```
Comment (cfsnook) [inline ac.soton.xeventb.xcontext/src/ac/soton/xeventb/xcontext/validation/XContextValidator.xtend:244]: could you use
((EventBObject)obj).getContaining(ContextPackage.Literals.CONTEXT);

**Review by cfsnook — COMMENTED** [review]
looks ok.. but see detailed comments

File: ac.soton.xeventb.xcontext/src/ac/soton/xeventb/xcontext/validation/XContextValidator.xtend (hunk 3 above)
Comment (tshoang) [reply ac.soton.xeventb.xcontext/src/ac/soton/xeventb/xcontext/validation/XContextValidator.xtend:247]: Thanks @cfsnook. I reimplemented the utility methods to use EventBObject.getContaining() method.

**Review by cfsnook — APPROVED** [review]

**Pull request merged**